README.md

```diff
@@ -1,13 +1,132 @@
 
-Installs a minimal Python3 environment for HEP machine learning
+Provides a minimal Python3 environment for machine learning focused towards HEP use cases
 
 [![License](https://img.shields.io/badge/License-BSD%203--Clause-blue.svg)](https://opensource.org/licenses/BSD-3-Clause)
 [![Build Status](https://travis-ci.com/matthewfeickert/HEPML-env.svg?branch=master)](https://travis-ci.com/matthewfeickert/HEPML-env)
 
+---
+
+Setting up a machine learning environment has gotten easier recently, but there are at times still problems that arise from time to time. HEPML-env allows for easily setting up a standard machine learning Python environment that should allow you to get to work with HEP data immediately. It should be machine agnostic, such that it can setup an identical environment on your laptop or on LXPLUS.
+
+### The default environment
+
+- [numpy](https://github.com/numpy/numpy)
+- [scipy](https://github.com/scipy/scipy)
+- [matplotlib](https://github.com/matplotlib/matplotlib)
+- [pandas](https://github.com/pandas-dev/pandas)
+- [h5py](https://github.com/h5py/h5py)
+- [uproot](https://github.com/scikit-hep/uproot)
+- [scikit-learn](https://github.com/scikit-learn/scikit-learn)
+- [scikit-image](https://github.com/scikit-image/scikit-image)
+- [tensorflow](https://github.com/tensorflow/tensorflow)
+- [jupyter](https://github.com/jupyter)
+
+
+### Core features
+- Reproducibility
+- Portability
+- ROOT-less
+
+## Requirements
+
+- GNU
+    - gcc/g++ (5.4 or higher is recommended)
+    - zlibc
+    - zlib1g-dev
+    - libssl-dev
+    - wget
+    - make
+    - Bash
+- Python 3.6 or higher
+- [pipenv](https://docs.pipenv.org/)
+
+HEPML uses Python 3 and pipenv, but if you don't have them installed already the [`install.sh`](https://github.com/matthewfeickert/HEPML-env/blob/master/scripts/install.sh) will take care of that for you.
+
+<details>
+ <summary>Installing GNU requirements on Debian/Ubuntu</summary>
+
+On Debian/Ubuntu the GNU requirements can be met by
+
+```
+apt install gcc g++ git zlibc zlib1g-dev libssl-dev wget make
+```
+
+</details>
+
+## Installation and setup
+
+1. Clone the repo
+
+<details>
+ <summary>Note on cloning on LXPLUS</summary>
+
+When trying to use SSH with GitHub on LXPLUS it is important to make sure that your `~/.ssh/config` is properly configured. It may need to contain something along the lines of
+
+```
+Host github.com
+    IdentityFile ~/.ssh/id_rsa-github
+    IdentitiesOnly yes
+```
+
+</details>
+
+
+2. If the above requirements are not already satisfied, run the installer
+
+```
+bash scripts/install.sh
+```
+
+3. Make a directory for your project
+
+```
+mkdir ~/projects/HEPML_project
+cd ~/projects/HEPML_project
+```
+
+4. Copy the included `Pipfile` and `Pipfile.lock` to the project directory
+
+5. Install the environment with pipenv
+
+```
+# From inside the project directory
+pipenv install
+```
+
+6. Launch into the project environment with pipenv
+
+```
+pipenv shell
+```
+
+## Questions
```

Comment:
I might just be an old and weird person, but personally I sort of enjoy just shoving a few directories in my `PATH` and / or `LD_LIBRARY_PATH` in the `.bashrc` file and having only one environment. Is it completely insane to support that?

Reply:
Yeah it does sound reasonable: it's definitely true that a lot of people will shoot themselves in the foot if they try to do what I'm asking for. But (why stop the random violent analogies) we should give everyone enough rope to hang themselves too.

Reply:
What about if I added a "I just want to use pip" section to the Questions and then indicate that in the future a script will be added that a user can run to generate a `requirements.txt` from the `Pipfile`?

Reply:
Actually, no need for me to do any work. [Pipenv already does this](https://docs.pipenv.org/advanced/#generating-a-requirements-txt) with
```
pipenv lock --requirements
```

Reply:
Added in commit ecf425d3a59dbaa3f61f0b08f5d0a71cb7210451